### PR TITLE
Handle mixins with property values of inherit and initial

### LINKED ISSUE #3818: ApplyShim does not correctly handle values of `inherit` or `initial` in mixins
The apply shim uses `initial` as a special value to help it forward default values to properties. Consider:

```
.foo {
  border: 2px solid red;
  @apply --foo;
}
```

In this case, if `--foo` does not contain a border property, elements matching `.foo` will have a 2px red border. Under the apply shim, if one setting of `--foo` contains `border` and another does not, we ensure that all `--foo` declarations contain at least `border: initial`. This allows us to write a rule like this:

```
.foo {
  border: 2px solid red;
  /*  shim for @apply --foo; */
  border: var(--foo_-_border, 2px solid red);
}
```

This works *unless* the user wants to set a value of `initial` for `border` in `--foo`. In this case the value is incorrect. In the native case, the computed value is `initial` and in the shim'd case it is the fallback.

See: http://jsbin.com/veqebo/edit?html,output.

### PR DESCRIPTION
**Pull request by azakus**
Fixes #3818

### CODE REVIEW

File: src/lib/apply-shim.html [hunk 1]
```diff
@@ -102,19 +103,51 @@
...
+  function getInitialValueForProperty(property) {
+    if (!measureElement) {
+      measureElement = document.createElement('meta');
+      measureElement.style.all = 'initial';
+      document.head.appendChild(measureElement);
+    }
+    return window.getComputedStyle(measureElement).getPropertyValue(property);
+  }
+
+  function replaceInitialOrInherit(property, value) {
```
Comment (sorvell) [inline src/lib/apply-shim.html]: put on ApplyShim object so it's easier to override in the build.

File: test/unit/styling-cross-scope-apply.html [hunk 2]
```diff
@@ -811,6 +876,35 @@
     CustomElements.takeRecords();
     assertComputed(e, '20px');
   });
+
+  test('Mixins can set "inherit" for a property', function() {
+    if (navigator.userAgent.match(/Trident/)) {
```
Comment (sorvell) [inline test/unit/styling-cross-scope-apply.html]: Add a comment: "IE does not support `inherit`"

File: test/unit/styling-cross-scope-apply.html [hunk 3]
```diff
@@ -811,6 +876,35 @@
...
+      this.skip();
+    }
+    var e = document.createElement('x-inherit');
+    document.body.appendChild(e);
+    CustomElements.takeRecords();
+    assertComputed(e.$.child, 'rgb(0, 255, 0)', 'color');
+    if (!Polymer.Settings.useNativeCSSProperties) {
+      assertComputed(e.$.child, '2px');
+    } else {
+      // this is not supported for the apply shim
```
Comment (sorvell) [inline test/unit/styling-cross-scope-apply.html]: Let's be more explicit so we remember this: "Setting `inherit` on a non-inheriting property is not supported under the apply shim."

File: test/unit/styling-cross-scope-apply.html [hunk 4]
```diff
@@ -811,6 +876,35 @@
...
+      assert.throws(function() {
+        assertComputed(e.$.child, '2px');
+      })
+    }
+  });
+
+  test('Mixins can set "initial" for a property', function() {
+    if (navigator.userAgent.match(/Trident/)) {
```
Comment (sorvell) [inline test/unit/styling-cross-scope-apply.html]: Add a comment: "IE does not support `initial`"

File: test/unit/styling-cross-scope-apply.html (hunk 3 above)
Comment (azakus) [inline test/unit/styling-cross-scope-apply.html]: Done.

**Comment by sorvell** [comment]
LGTM

**Pull request merged**